### ISSUE TITLE
Editor text input cursor position resets on save

### ISSUE THREAD

**Issue by snoopdouglas** [godotengine/godot#45728]
**Godot version:**
3.2.4-rc1
nb. I have noticed this behaviour in 3.2.3 too.

**OS/device including version:**
Ubuntu Linux 20.04

**Issue description:**
When I'm editing a value in the node inspector and the scene is saved (either automatically or through me pressed Ctrl+S), the cursor goes to the beginning of the line. This may be while I'm in the middle of typing, in which case, it feels pretty jarring.

This is possibly related to #20708.

**Steps to reproduce:**
* Edit a node's property
* Press Ctrl+S
* Notice the cursor snap to home

**Minimal reproduction project:**
n/a

**Comment by Sirignus** [comment]
This might be only related to [Ubuntu] Linux. (I'm on Windows 10)

I tried to reproduce this but did not succeed.

Most integers values you can edit prevent me from saving until I click elsewhere (Transform Position.x for instance)

But I could save without having my cursor go back to the start of the line when I edited a Text property from a RichTextLabel Control Node or in the filter properties panel. My guess so far is that you can only Ctrl+S when editing strings in the inspector.

That being said, I'm not sure about the autosave feature you're using. Is this the Editor Settings > Text Editor > Files >Auto-Save Intervals Seconds ? Even then I did try this with a gdscript and it worked fine (No cursor change happening) but since it doesn't auto save scenes I'm guessing this is not the one. (I tried to find an auto-save feature for scenes but could'nt).

I also tried this on Godot 3.2.3 stable and don't have this issue.

I'm not entirely sure I understand the problem though so if I'm missing something, please correct me.

**Comment by Calinou** [comment]
@snoopdouglas Do you have **Trim Trailing Whitespace** enabled in the Editor Settings?

**Comment by snoopdouglas** [comment]
@Sirignus I've just noticed something weird that might narrow this down: it only seems to be happening on my main scene.

Additionally, yes, it isn't possible to save when in the middle of editing integers or floats. I first noticed it when editing an exported string, but it also happens when editing a node's 'editor description'.

**Comment by snoopdouglas** [comment]
@Calinou Yes, I have got that enabled - but disabling it isn't stopping this from happening.

**Comment by snoopdouglas** [comment]
Apologies for 3 comments in a row, but I've _have_ actually narrow this down now. This is only happening for nodes which have scripts attached to them.

I'll update my initial description.

**Comment by Sirignus** [comment]
I was missing the script attached indeed. I can now reproduce this bug with either 3.2.4.rc and 3.2.3 stable on Windows 10.

While trying to reproduce this on master branch and I noticed something interesting. When your script attached to the Node has errors, the bug is missing but whenever you fix your gdscript the bug reappears.

I'm guessing the error in the script prevents something from happening which is most likely the cause of this bug.

This is reproduceable in 3.2.3 stable and 3.2.4.rc.

It mostly tells us something we already knew (script attached related) but might narrow down the search for a bugfix.
